apps/dashboard/src/hooks/use-sync-workflow.tsx

```diff
@@ -0,0 +1,213 @@
+              >
+                Switch to production
+                <RiArrowRightSLine />
+              </Button>
+            </div>
+          </div>
+          <ToastClose onClick={close} />
+        </>
+      ),
+      options: {
+        position: 'bottom-right',
+      },
+    });
+  };
+
+  const onSyncError = (error: unknown) => {
+    toast.dismiss(loadingToast);
+    setIsLoading(false);
+
+    toast.error('Failed to sync workflow', {
+      description: error instanceof Error ? error.message : 'There was an error syncing the workflow.',
+    });
+  };
+
+  const { mutateAsync: syncWorkflowMutation } = useMutation({
+    mutationFn: async () =>
+      syncWorkflow(workflow._id, { targetEnvironmentId: getTargetEnvironment(currentEnvironment)?._id || '' }).then(
+        (res) => res.data
+      ),
+    onMutate: () => {
+      setIsLoading(true);
+      loadingToast = toast.loading('Syncing workflow...');
+    },
+    onSuccess: onSyncSuccess,
+    onError: onSyncError,
+  });
+
+  const ConfirmationModal = () => {
+    async function onConfirm() {
+      setShowConfirmModal(false);
+      await syncWorkflowMutation();
+    }
+
+    return (
+      <Dialog modal open={showConfirmModal} onOpenChange={setShowConfirmModal}>
+        <DialogPortal>
+          <DialogOverlay />
+          <DialogContent className="sm:max-w-[440px]">
+            <div className="flex items-start gap-4 self-stretch">
+              <div className="bg-warning/10 flex items-center justify-center gap-2 rounded-[10px] p-2">
+                <RiAlertFill className="text-warning size-6" />
+              </div>
+              <div className="flex flex-1 flex-col items-start gap-1">
+                <DialogTitle className="text-md font-medium">
+                  Sync workflow to {getTargetEnvironment(currentEnvironment)?.name}
+                </DialogTitle>
+                <DialogDescription className="text-foreground-600">
+                  Workflow already exists in {getTargetEnvironment(currentEnvironment)?.name}. Proceeding will overwrite
+                  the existing workflow.
+                </DialogDescription>
+              </div>
+            </div>
+            <DialogFooter>
+              <DialogClose asChild aria-label="Close">
+                <Button type="button" size="sm" variant="outline" onClick={() => setShowConfirmModal(false)}>
+                  Cancel
+                </Button>
+              </DialogClose>
+              <DialogClose asChild aria-label="Close">
+                <Button type="button" size="sm" variant="primary" onClick={onConfirm}>
+                  Proceed
+                </Button>
+              </DialogClose>
+            </DialogFooter>
+          </DialogContent>
+        </DialogPortal>
+      </Dialog>
+    );
+  };
+
+  return {
+    safeSync,
+    sync: syncWorkflowMutation,
+    isSyncable,
+    isLoading,
+    tooltipContent: getTooltipContent(),
```

Comment:
This hook is overloaded with headless and UI logic. TooltipContent and ConfirmationModal need to be transformed to components and be removed from the hook.

Reply:
I can understand moving `ConfirmationModal` to separate component although I would argue that the confirmation modal is closely coupled just with the promotion logic and will be used only with this hook and nowhere else, so it should be either in one file or in one folder e.g. `/hooks/sync/*`

But why `TooltipContent`? Its not component, it more similar to an error message and everyone consuming this hook should show this very same message across the app so we are consistent.

Reply:
Moved the UI logic to separate components and simplified the hook by extracting environment logic to `useEnvironment` hook